airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -481,13 +485,21 @@ def request_params(
 
         stream_date = stream_state.get(self.cursor_field)
         next_token = (next_page_token or {}).get("next_token")
+        primary_key = (next_page_token or {}).get("primary_key")
         start_date = next_token or stream_date or self.start_date
+        self.prev_start_date = start_date
 
         query = f"SELECT {','.join(selected_properties.keys())} FROM {self.name} "
         if start_date:
-            query += f"WHERE {self.cursor_field} >= {start_date} "
+            if primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
+                query += f"WHERE ({self.cursor_field} = {start_date} AND {self.primary_key} > '{primary_key}') OR ({self.cursor_field} > {start_date}) "
+            else:
+                query += f"WHERE {self.cursor_field} >= {start_date} "
```

Comment:
nit 
```suggestion
           query += f"WHERE {self.cursor_field} >= {start_date} "
            if primary_key and self.name not in UNSUPPORTED_FILTERING_STREAMS:
                query += f" AND {self.primary_key} > '{primary_key}') OR ({self.cursor_field} > {start_date}) "
```

Reply:
no, it cannot be changed such way:

first query starts:
```
{self.cursor_field} = {start_date}
```

2-nd query starts:
```
{self.cursor_field} >= {start_date}
```